clients/src/main/java/org/apache/kafka/common/requests/AbstractResponse.java

```diff
@@ -47,9 +48,9 @@ public ByteBuffer serialize(short version, ResponseHeader responseHeader) {
         return Collections.singletonMap(error, 1);
     }
 
-    protected Map<Errors, Integer> errorCounts(Map<?, Errors> errors) {
+    protected Map<Errors, Integer> errorCounts(Collection<Errors> errors) {
```

Comment:
In addition to having a function which takes a `Collection<Errors>`, it would be useful to have a function which takes a `Stream<Errors>`.  That way we could avoid materializing the list when all we want to do is count the number of errors of each type.

Reply:
Do we have any callers that would use this? If not, then I suggest we consider it when we need it.

Reply:
I see that we have a few. Let's keep this for a separate PR though, this one is pretty large as it is.

Reply:
OK.  I filed [KAFKA-8955 Add an AbstractResponse#errorCounts method that takes a stream or iterable](https://issues.apache.org/jira/browse/KAFKA-8955) to track this.